src/muse/demand_matching.py

```diff
@@ -264,7 +264,7 @@ def _demand_matching_impl(
     dimensions have coordinates. Input sanitization is performed in `demand_matching`
     proper.
     """
-    from numpy import isnan, prod
+    from numpy import all, isnan, prod
```

Comment:
Why are there changes in this PR about demand matching and timeslices when it is supposed to be only about the version?